java-squid/src/main/java/org/sonar/java/se/checks/NullDereferenceCheck.java

```diff
@@ -102,10 +103,10 @@ public void checkPostStatement(CheckerContext context, Tree syntaxNode) {
       assert val != null && val.equals(SymbolicValue.NULL_LITERAL);
     } else if (syntaxNode.is(Tree.Kind.METHOD_INVOCATION)) {
       if (isAnnotatedCheckForNull((MethodInvocationTree) syntaxNode)) {
-        return Lists.newArrayList(
-            val.setConstraint(context.getState(), ConstraintManager.NullConstraint.NULL),
-            val.setConstraint(context.getState(), ConstraintManager.NullConstraint.NOT_NULL)
-        );
+        List<ProgramState> states = new ArrayList<>();
+        states.addAll(val.setConstraint(context.getState(), ConstraintManager.NullConstraint.NULL));
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'val' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)